### PR TITLE
Update circleci config, add badges

### LINKED ISSUE #148: Testing & quality tools
**Is your feature request related to a problem? Please describe.**
It is not really a feature, I would like to suggest adding a CI, some go quality tools and a coverage tool could increase the quality and ease the maintenance.

**Describe the solution you'd like**
For instance using https://circleci.com as CI, https://golangci.com as quality check tool and https://codecov.io as a coverage tool, they are all free for open source project.

**Describe alternatives you've considered**
/

**Additional context**
/

### PR DESCRIPTION
**Pull request by antham**
Fix #148

### CODE REVIEW

**Comment by dawidd6** [comment]
To be easily buildable i guess.

**Comment by antham** [comment]
I don't get the point, in which way it is more complicated to build with dep ?

**Comment by dawidd6** [comment]
To build with dep, first you need to have dep and this is additional dependency, which is IMO not necessarily needed. Vendor directory is very convenient.

**Comment by antham** [comment]
Ok right. So it means that dep has not to be used in circleci to follow the same path.

**Comment by jesseduffield** [comment]
@antham I've added the GITHUB_TOKEN env variable to the lazygit circle ci project, and I've turned on the setting for making builds for PRs.

To confirm: in order to trigger a deployment, I need to push a tag to e.g. the master branch?

**Comment by antham** [comment]
Ok so I see building is working for PR's with my rebase and codecov is working as well.

Yes you would have to create a tag against master, you can do it directly from github interface when you create a release it creates a tag it's more convenient.

**Comment by jesseduffield** [comment]
cool, so when I git publish release here:

image

it will trigger a build on CircleCI which will do a deploy as its final step? Sounds good to me

My only remaining reservation is that currently we have a VERSION file in the project's root directory that I'm manually bumping with each new release, in case somebody has installed the app via go install, so that we can still see last release version number when they raise an issue. Should we scrap that? Maybe we can display a commit SHA at the bottom right in place of a version number for go-users.

**Comment by antham** [comment]
Yes if it works properly it will, I just tried locally so we have to see in production.

What do you mean when you say

> Maybe we can display a commit SHA at the bottom right in place of a version number for go-users.

You are still talking about the VERSION file ?

If you just state an explicit command to do in bug reporting, don't you think it would be enough ? When I open a bug it's not describe to dump the version written in VERSION file.

**Comment by jesseduffield** [comment]
@antham yeah that sounds reasonable, I'll remove the VERSION file cos it looks like we don't need it

**Comment by antham** [comment]
@jesseduffield I updated the github issue template please have a look.

**Comment by jesseduffield** [comment]
@antham maybe just specify `in the lazygit directory` so that we don't get people giving us _their_ last commit SHA from their own repos

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,26 +1,51 @@
...
+      - image: circleci/golang:1.10
 
-      # Specify service dependencies here if necessary
-      # CircleCI maintains a library of pre-built images
-      # documented at https://circleci.com/docs/2.0/circleci-images/
-      # - image: circleci/postgres:9.4
+    working_directory: /go/src/github.com/jesseduffield/lazygit
+    steps:
+      - checkout
+      - run:
```
[the remark this answers is not in the document]
Comment (jesseduffield) [reply .circleci/config.yml]: whoops I meant the lines beneath:
```
          name: Install dep
          command: |
            go get -u github.com/golang/dep/cmd/dep
      - run:
          name: Install vendors
          command: |
            dep ensure -v
```

**Comment by jesseduffield** [comment]
@antham is this PR ready for re-review?

**Comment by antham** [comment]
@jesseduffield Yes you can

**Comment by jesseduffield** [comment]
Looks good, merging!

**Pull request merged**